payment-element/server/dotnet/Program.cs

```diff
@@ -1,30 +1,97 @@
+    Url = "https://github.com/stripe-samples",
+    Version = "0.1.0",
+};
+
+StripeConfiguration.ApiKey = Environment.GetEnvironmentVariable("STRIPE_SECRET_KEY");
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.Configure<StripeOptions>(options =>
+{
+    options.PublishableKey = Environment.GetEnvironmentVariable("STRIPE_PUBLISHABLE_KEY");
+    options.SecretKey = Environment.GetEnvironmentVariable("STRIPE_SECRET_KEY");
+    options.WebhookSecret = Environment.GetEnvironmentVariable("STRIPE_WEBHOOK_SECRET");
+});
+
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment())
+{
+    app.UseDeveloperExceptionPage();
+}
+
+app.UseStaticFiles(new StaticFileOptions()
 {
-    public class Program
+    FileProvider = new PhysicalFileProvider(
+        Path.Combine(Directory.GetCurrentDirectory(),
+        Environment.GetEnvironmentVariable("STATIC_DIR"))
+    ),
+    RequestPath = new PathString("")
+});
+
+app.MapGet("/config", (IOptions<StripeOptions> options) => new { options.Value.PublishableKey });
+
+app.MapPost("/create-payment-intent", async () =>
+{
+    try
     {
-        public static void Main(string[] args)
+        var service = new PaymentIntentService();
+        var paymentIntent = await service.CreateAsync(new PaymentIntentCreateOptions
         {
-            CreateHostBuilder(args).Build().Run();
-        }
-
-        public static IHostBuilder CreateHostBuilder(string[] args)
-        {
-            DotNetEnv.Env.Load();
-            return Host.CreateDefaultBuilder(args)
-                .ConfigureWebHostDefaults(webBuilder =>
-                {
-                    webBuilder.UseStartup<Startup>();
-                    webBuilder.UseWebRoot(Environment.GetEnvironmentVariable("STATIC_DIR"));
-                });
-        }   
+            Amount = 1999,
+            Currency = "EUR",
+            AutomaticPaymentMethods = new PaymentIntentAutomaticPaymentMethodsOptions
+            {
+                Enabled = true,
+            },
+        });
+
+        return Results.Ok(new { ClientSecret = paymentIntent.ClientSecret });
     }
-}
+    catch (StripeException e)
+    {
+        return Results.BadRequest(new { error = new { message = e.StripeError.Message } });
+    }
+});
+
+app.MapPost("/webhook", async (HttpRequest request, IOptions<StripeOptions> options) =>
+{
+    var json = await new StreamReader(request.Body).ReadToEndAsync();
+    Event stripeEvent;
+    try
+    {
+        stripeEvent = EventUtility.ConstructEvent(
+            json,
+            request.Headers["Stripe-Signature"],
+             options.Value.WebhookSecret
+        );
+        app.Logger.LogInformation($"Webhook notification with type: {stripeEvent.Type} found for {stripeEvent.Id}");
+    }
+    catch (Exception e)
+    {
+        Console.WriteLine($"Something failed {e}");
+        return Results.BadRequest();
+    }
+
+    if (stripeEvent.Type == "checkout.session.completed")

```

Comment:
For the PaymentElement integration, we want to listen for `payment_intent.succeeded` 

We may have missed this in other server implementations here. 

```cs
var paymentIntent = stripeEvent.Data.Object as Stripe.PaymentIntent;
```

I think we might also have constants for the event types in stripe-dotnet: https://github.com/stripe/stripe-dotnet/blob/master/src/Stripe.net/Constants/Events.cs#L558

Might be able to say something like:

```cs
if (stripeEvent.Type == Stripe.Events.PaymentIntentSucceeded)
```

Reply:
Updated